main/webapp/modules/core/scripts/facets/facet.js

```diff
@@ -67,3 +110,20 @@ class Facet {
   dispose() {
   };
 };
+
+Facet.references = {
+  "row.starred": 
+    { source: "<stars>", label: 'core-views/starred-rows' },
+  "row.flagged": 
+    { source: "<flags>", label: 'core-views/flagged-rows' },
+  "(filter(row.columnNames,cn,isNonBlank(cells[cn].value)).length()==0).toString()": 
+    { source: "<blank-rows>", label: 'core-views/blank-rows' },
+  "filter(row.columnNames,cn,isBlank(cells[cn].value))": 
+    { source: "<blank-values>", label: 'core-views/blank-values' },
+  "filter(row.columnNames,cn,isBlank(if(row.record.fromRowIndex==row.index,row.record.cells[cn].value.join(\"\"),true)))": 
+    { source: "<blank-records>", label: 'core-views/blank-records' },
+  "filter(row.columnNames,cn,isNonBlank(cells[cn].value))": 
+    { source: "<non-blank-values>", label: 'core-views/non-blank-values' },
+  "filter(row.columnNames,cn,isNonBlank(if(row.record.fromRowIndex==row.index,row.record.cells[cn].value.join(\"\"),null)))": 
+    { source: "<non-blank-records>", label: 'core-views/non-blank-records' }

```

Comment:
I would rather store this mapping in the definition of the facets themselves (in the menu). Because it is likely that people forget to update this when they add a new facet in that menu, or for instance when an extension registers a facet there. I think this could be done simply changing the method which adds a new facet, so that the facet name can be set directly there.